hack/gatewayapi-conformance.sh

```diff
@@ -32,22 +34,30 @@ cd "${CLONE_DIR}"
 # find the branch of gateway-api repo
 RELEASE_VERSION=$(\grep -oP "^\d+\.\d+" <<<"${BUNDLE_VERSION#v}")
 BRANCH="release-${RELEASE_VERSION}"
-echo "gateway-api repo branch: ${BRANCH}"
+echo "gateway-api repo branch \"${BRANCH}\" into ${CLONE_DIR}..."
 
 # clone the branch that matched the installed CRDs version
 # branch release-1.0 -> bundle-version v1.0.0
 # branch release-1.1 -> bundle-version v1.1.1
 # branch release-1.2 -> bundle-version v1.2.1
 git clone --branch "${BRANCH}" https://github.com/kubernetes-sigs/gateway-api
 cd gateway-api
+
+if [[ "$BUNDLE_VERSION" = "v1.2.1" ]]; then
```

Comment:
_technically_ you're going to want to cover anything in `v1.2.x`. Hopefully we don't have any more patches in `v1.2`, but you never know.

Reply:
Ah, I hadn't seen that the proposed backport at https://github.com/kubernetes-sigs/gateway-api/pull/3688 had been closed.  

However, I'm ambivalent about changing this, knowing that there's still a chance that there could be a v1.2.2 with the patch, or a v1.2.2 with some other change that causes the cherry-pick to fail.  

If we bump to v1.2.2+ and find that the cherry-pick is still needed, we can update this test then with the knowledge that it remains applicable and necessary.